src/libstd/sys/unix/net.rs

```diff
@@ -33,6 +33,14 @@ use libc::SOCK_CLOEXEC;
 #[cfg(not(target_os = "linux"))]
 const SOCK_CLOEXEC: c_int = 0;
 
+// Another conditional contant for name resolution: Macos et iOS use
+// SO_NOSIGPIPE as a setsockopt flag to disable SIGPIPE emission on socket.
+// Other platforms do otherwise.
+#[cfg(target_vendor = "apple")]
+use libc::SO_NOSIGPIPE;
```

Comment:
> Android is covered by target_os="linux",

To the best of my knowledge its not true, because value of `target_os` on Android is `android`, not `linux`.

Blind is good enough; better than feature disparity IMO.